src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -184,7 +185,11 @@ public UserDefinedFunction WithBinding(INameResolver nameResolver, IBinderGlue b
             return func;
         }
 
-        private static readonly ISet<string> _restrictedUDFNames = new HashSet<string> { "Type", "IsType", "AsType" };
+        private static readonly ISet<string> _restrictedUDFNames = new HashSet<string>
+        {
+            "Type", "IsType", "AsType", "Set", "Collect",
+            "ClearCollect", "UpdateContext", "Navigate", "Select", "Reset",
+        };
 
```

Comment:
Can we make this list attribute based on TexlFunction level? And then filter it using that property of that function instead of static list?

Reply:
There is something nice about seeing the list outright here. 
Even if we do get rid of this list from the product, it would be good to have this explicit list in the tests so that we can verify. 

It would also be good if the analysis piece used this same list and enforced things.

Reply:
Adding it in tests make sense but in product this needs to be property based on TexlFunction to be more explicit.

Reply:
☝️ We should not be hardcoding the function names like this. This should be a property of the function itself, so it can be queried from other places such as in analysis.

Reply:
Wouldn't we still end up some list here? I see atleast one possibility which is `Type` and it is not a TexlFunction .

Reply:
Good point on Type - it's not a function. Is it a keyword now that we have UDTs? If so, should we forbid all keywords as well?